### PR TITLE
[MODULAR] The start of Medigun changes

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
This is going to be the first PR out of many from me to try and address the issues with the Medigun in it's current state. I am going to focus on improving the gun and giving it it's own identity before I add more guns.

First off, the ability to heal corpses is being removed from the medigun cells. The ability to heal corpses before was restricted to surgery or more limited resource healing tools like chems/botany stuff, having it on a rechargeable gun is too much power.  Later on, I will be implementing cells that make revival easier. The ability to heal synths and borgs is also patched out with this PR

Second, the power consumption of the medigun shots have been doubled along with the amount of healing that each shot gives being increased, because of this a slight delay between shots has been added. Spamming the medigun sucks for both the user and the patient, this should hopefully make the gun feel better to use. This is also a nerf to the brute and burn modes, reducing their healing by 25%

Lastly, the Medigun crates from cargo no longer come with cells. This means that either the consumer has to spend 1000 more credits on the purchase or interact with medical/research to get the cells. This should hopefully make the gun a bit harder to obtain for non-medical roles. 

## Why It's Good For The Game
Balances the Medigun and gives it a better feel.

## Changelog
:cl:
balance: Mediguns can no longer heal corpses
balance: Mediguns shots are more potent but also take up more energy. 
balance: The Medigun kit from cargo no longer comes with cells included. 
fix: Medigun shots can no longer heal borgs and synths
/:cl:

### CODE REVIEW

**Comment by softcerv** [comment]
> 
> 
> You currently have a negative Fix/Feature pull request delta of -1. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

cease

**Comment by SpaceVampire** [comment]
Honestly, after thinking about it for a bit and having seen it in action.
I believe that the Medigun is a fundamentally bad idea.
It is just healing with zero downsides.
That works on everything.
With no thought at all needed.
It is going to end up in the kind of balance zone.
Either it's better than everything else, and never a reason not to use it.
Or it's worse, and never a reason to use it.
SmartDarts from the oldbase had a much better design.
Because you had to actually work with chemistry to fill them up.
Instead of just using a weapon that bypasses all of the systems.

**Comment by softcerv** [comment]
> 
> 
> Honestly, after thinking about it for a bit and having seen it in action.
> I believe that the Medigun is a fundamentally bad idea.
> It is just healing with zero downsides.
> That works on everything.
> With no thought at all needed.
> It is going to end up in the kind of balance zone.
> Either it's better than everything else, and never a reason not to use it.
> Or it's worse, and never a reason to use it.
> SmartDarts from the oldbase had a much better design.
> Because you had to actually work with chemistry to fill them up.
> Instead of just using a weapon that bypasses all of the systems.

The healing functionality on this working on being toned down, there are going to be things to discourage doctors from the brute and burn modes. I am planning on shifting it from a raw healing tool to more of a general medical tool by adding more cells that are focused on utility rather than raw healing. If you would like to hear about some of the things that me and other have in mind for this, please check the medigun discussion thread on discord.

File: modular_skyrat/modules/cargo/code/packs.dm [hunk 1]
```diff
@@ -90,16 +90,16 @@
 	crate_name = "compact defibrillator crate"
 
 /datum/supply_pack/medical/medigun
-	name = "CWM-479 Medigun Starter Kit"
-	desc = "Contains a single VeyMedical CWM-479 model medical gun with basic cells"
+	name = "CWM-479 Medigun"
+	desc = "Contains a single VeyMedical CWM-479 model medical gun. Cells not included"
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/cargo/code/packs.dm]:
```suggestion
	desc = "Contains a single VeyMedical CWM-479 model medical gun; cells not included."
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 2]
```diff
@@ -35,7 +36,10 @@
 
 /obj/projectile/energy/medical/default/on_hit(mob/living/target)
 	.=..()
-	target.adjustOxyLoss(-5)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustOxyLoss(-10)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustOxyLoss(-10)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 3]
```diff
@@ -35,7 +36,10 @@
 
 /obj/projectile/energy/medical/default/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 4]
```diff
@@ -49,7 +53,10 @@
 
 /obj/projectile/energy/medical/brute1/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 5]
```diff
@@ -49,7 +53,10 @@
...
-	target.adjustBruteLoss(-5)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustBruteLoss(-7.5)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustBruteLoss(-7.5)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 6]
```diff
@@ -61,7 +68,10 @@
 
 /obj/projectile/energy/medical/burn1/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 7]
```diff
@@ -61,7 +68,10 @@
...
-	target.adjustFireLoss(-5)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustFireLoss(-7.5)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustFireLoss(-7.5)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 8]
```diff
@@ -73,8 +83,11 @@
 
 /obj/projectile/energy/medical/toxin1/on_hit(mob/living/target)
 	.=..()
-	target.adjustToxLoss(-2.5)
-	target.radiation = max(target.radiation - 20, 0)//Toxin is treatable, but inefficent//
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustToxLoss(-5)
+		target.radiation = max(target.radiation - 40, 0)//Toxin is treatable, but inefficent//
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustToxLoss(-5)
	target.radiation = max(target.radiation - 40, 0)//Toxin is treatable, but inefficent//
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 9]
```diff
@@ -73,8 +83,11 @@
 
 /obj/projectile/energy/medical/toxin1/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 10]
```diff
@@ -88,7 +101,10 @@
 
 /obj/projectile/energy/medical/upgraded/brute2/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 11]
```diff
@@ -88,7 +101,10 @@
...
-	target.adjustBruteLoss(-7.5)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustBruteLoss(-11.25)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustBruteLoss(-11.25)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 12]
```diff
@@ -100,7 +116,10 @@
 
 /obj/projectile/energy/medical/upgraded/burn2/on_hit(mob/living/target)
 	.=..()
-	target.adjustFireLoss(-7.5)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustFireLoss(-11.25)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustFireLoss(-11.25)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 13]
```diff
@@ -100,7 +116,10 @@
 
 /obj/projectile/energy/medical/upgraded/burn2/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 14]
```diff
@@ -111,7 +130,10 @@
 
 /obj/projectile/energy/medical/upgraded/oxy2/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 15]
```diff
@@ -111,7 +130,10 @@
...
-	target.adjustOxyLoss(-10)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustOxyLoss(-20)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustOxyLoss(-20)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 16]
```diff
@@ -123,8 +145,11 @@
 
 /obj/projectile/energy/medical/upgraded/toxin2/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 17]
```diff
@@ -123,8 +145,11 @@
...
-	target.adjustToxLoss(-3.5)
-	target.radiation = max(target.radiation - 40, 0)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustToxLoss(-7.5)
+		target.radiation = max(target.radiation - 60, 0)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustToxLoss(-7.5)
	target.radiation = max(target.radiation - 60, 0)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 18]
```diff
@@ -137,7 +162,10 @@
 
 /obj/projectile/energy/medical/upgraded/brute3/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 19]
```diff
@@ -137,7 +162,10 @@
...
-	target.adjustBruteLoss(-10)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustBruteLoss(-15)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustBruteLoss(-15)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 20]
```diff
@@ -149,7 +177,10 @@
 
 /obj/projectile/energy/medical/upgraded/burn3/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 21]
```diff
@@ -149,7 +177,10 @@
...
-	target.adjustFireLoss(-10)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustFireLoss(-15)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustFireLoss(-15)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 22]
```diff
@@ -160,7 +191,10 @@
 
 /obj/projectile/energy/medical/upgraded/oxy3/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 23]
```diff
@@ -160,7 +191,10 @@
...
-	target.adjustOxyLoss(-15)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustOxyLoss(-30)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustOxyLoss(-30)
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 24]
```diff
@@ -172,7 +206,10 @@
 
 /obj/projectile/energy/medical/upgraded/toxin3/on_hit(mob/living/target)
 	.=..()
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	. = ..()
```

File: modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm [hunk 25]
```diff
@@ -172,7 +206,10 @@
...
-	target.adjustToxLoss(-5)
-	target.radiation = max(target.radiation - 60, 0)
+	if(target.stat == DEAD)
+		return
+	else
+		target.adjustToxLoss(-5)
+		target.radiation = max(target.radiation - 80, 0)
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/modular_weapons/code/modules/projectiles/guns/energy/medgun/medcells.dm]:
```suggestion
	target.adjustToxLoss(-5)
	target.radiation = max(target.radiation - 80, 0)
```

**Comment by ZephyrTFA** [comment]
Meant to do a code review; did individual comments instead. Sorry about the emails.

**Comment by softcerv** [comment]
> 
> 
> Meant to do a code review; did individual comments instead. Sorry about the emails.

It's all good, thank you a bunch for looking over my code by the way!

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**